compiler-plugin/src/main/java/io/ballerina/c2c/tasks/C2CCodeGeneratedTask.java

```diff
@@ -69,25 +99,147 @@ public void perform(CompilerLifecycleEventContext compilerLifecycleEventContext)
         if (cloud == null || !KubernetesUtils.isBuildOptionDockerOrK8s(cloud)) {
             return;
         }
+
+        BalCommand balCommand = compilerLifecycleEventContext.balCommand();
         Optional<Path> executablePath = compilerLifecycleEventContext.getGeneratedArtifactPath();
         final Package currentPackage = compilerLifecycleEventContext.currentPackage();
-        PackageDescriptor descriptor = currentPackage.descriptor();
-        executablePath.ifPresent(path -> {
-            String executableJarName = "$anon".equals(descriptor.org().value()) ? path.getFileName().toString() :
-                    descriptor.org().value() + "-" + descriptor.name().value() +
-                            "-" + descriptor.version().value() + ".jar";
-            String outputName = "$anon".equals(descriptor.org().value()) ? extractJarName(path.getFileName()) :
-                    descriptor.name().value().toLowerCase(Locale.ROOT);
-            dataHolder.setOutputName(outputName);
-            addDependencyJars(compilerLifecycleEventContext.compilation(), executableJarName);
-            dataHolder.setSourceRoot(executablePath.get().getParent()
-                    .getParent().getParent());
-            codeGeneratedInternal(KubernetesUtils.getProjectID(currentPackage),
-                    path, compilerLifecycleEventContext.currentPackage());
+
+        if (balCommand == BalCommand.BUILD) {
+            PackageDescriptor descriptor = currentPackage.descriptor();
+            executablePath.ifPresent(path -> {
+                String executableJarName = "$anon".equals(descriptor.org().value()) ? path.getFileName().toString() :
+                        descriptor.org().value() + "-" + descriptor.name().value() +
+                                "-" + descriptor.version().value() + ".jar";
+                String outputName = "$anon".equals(descriptor.org().value()) ? extractJarName(path.getFileName()) :
+                        descriptor.name().value().toLowerCase(Locale.ROOT);
+                dataHolder.setOutputName(outputName);
+                addDependencyJars(compilerLifecycleEventContext.compilation(), executableJarName);
+                dataHolder.setSourceRoot(executablePath.get().getParent()
+                        .getParent().getParent());
+                codeGeneratedInternal(KubernetesUtils.getProjectID(currentPackage),
+                        path, compilerLifecycleEventContext.currentPackage(), balCommand);
+            });
+        } else {
```

Comment:
We can explicitly check for BalCommand.TEST here

Reply:
Added in [e332a9e](https://github.com/ballerina-platform/module-ballerina-c2c/pull/765/commits/e332a9ea20271f7c481c3750bad97af0f5f32df1)